static/js/pivotal.js

```diff
@@ -42,172 +40,212 @@
+    // When reset button clicked add Get stories button
+    $('.refresh').click(function() {
+      $(this).closest('.project').find('.story').html(button);
     });
 
-    // "get stories" button onClick handler
-    $('.getStories').click(pivotalButtonOnClickHandler);
-
-    function pivotalButtonOnClickHandler(e) {
+    $('.project').on('click', '.getStories', function(e) {
       e.preventDefault();
 
       var $this_button = $(e.currentTarget);
-      var project = $this_button.parents(config.selectors.project).data('id');
-      var failure_message = 'No stories found.';
-
-      diffs = getProjectDiffs();
+      var project = $this_button.parents('.project').data('id');
+      var diffs = getProjectDiffs();
 
-      $this_button.hide(); // do not show button
+      $this_button.hide();
 
       if (project in diffs) {
-        // great! we found diffs for this project; load the pivotal stories
         $this_button.siblings('.loading').show();
 
-        (function(project) {
-          var url = diffs[project];
-          // hashes: currentCommit...latestCommit
-          var hashes = (url.substr(url.lastIndexOf('/') + 1)).split('...');
-          var proj_name = url.match(config.github.url_compare_regexp)[1];
-
-          getGithubPivotalIDs(GITHUB_TOKEN, proj_name, hashes[0], hashes[1], function(commits) {
-            var messages = [];
-            for (commit in commits) {
-              messages.push(commits[commit]['commit']['message']); // get all github commit messages
-            }
-
-            var pivotal_ids = getCommitIDs(messages); // extract pivotal ticket IDs
-            if (!pivotal_ids) {
-              $this_button.parents(config.selectors.story_column).text(failure_message);
-              return;
-            };
-
-            $this_button.siblings('.loading').hide();
-
-            var display_message = [];
-            for (pt_id in pivotal_ids) {
-              var id = pivotal_ids[pt_id];
-
-              getPivotalInfo(PIVOTAL_TOKEN, id, function(info) {
-                display_message[id] = info;
-                injectPivotalStatus(project, display_message);
-              });
-            }
+        var url = diffs[project];
+        // hashes: currentCommit...latestCommit
+        var commitHashes = (url.substr(url.lastIndexOf('/') + 1)).split('...');
+        var repositoryName = url.match(/.*\/(.*)\/compare/)[1];
+
+        getGithubCommits(GITHUB_TOKEN, repositoryName, commitHashes, function(commits) {
+          // Array of comitt messages
+          var messages = commits.map(function(obj) {
+            return obj.commit.message;
           });
-        })(project);
+          // Array of pivotal story IDs
+          var pivotal_ids = getPivotalStoryIDs(messages);
+          if (!pivotal_ids) {
+            showNoStoriesMessage($this_button);
+            return;
+          };
+
+          var storyList = [];
+          for (var i = 0, imax = pivotal_ids.length; i < imax; i++) {
```

Comment:
I wanted to use that `imax` couple of lines beneath in the if statement `if (storyList.length === pivotal_ids.length) {` so changed the code to use `imax` there :pray: